pkg/controllers/client.go

```diff
@@ -86,6 +98,10 @@ func createSyncDeployment(c *daemon.Cluster) error {
 			return err
 		}
 	}
+
+	// delete the SyncConfigDeployment after the cluster is deployed.
```

Comment:
using channel to get delete result (true or false) and log the reason if delete the deployment failed.